Applications/spire/source/ui/properties_window_buttons_widget.cpp

```diff
@@ -71,27 +79,20 @@ properties_window_buttons_widget::properties_window_buttons_widget(
   apply_to_all_button->connect_clicked_signal(
     [=] { m_apply_to_all_signal(); });
   right_layout->addWidget(apply_to_all_button, 0, 1);
-  auto cancel_button = new flat_button(tr("Cancel"));
-  cancel_button->setFont(generic_button_font);
-  cancel_button->set_style(generic_button_style);
-  cancel_button->set_hover_style(generic_button_hover_style);
-  cancel_button->set_focus_style(generic_button_focus_style);
-  cancel_button->connect_clicked_signal([=] { m_cancel_signal(); });
-  right_layout->addWidget(cancel_button, 1, 1);
-  auto apply_button = new flat_button(tr("Apply"));
-  apply_button->setFont(generic_button_font);
-  apply_button->set_style(generic_button_style);
-  apply_button->set_hover_style(generic_button_hover_style);
-  apply_button->set_focus_style(generic_button_focus_style);
-  apply_button->connect_clicked_signal([=] { m_apply_signal(); });
-  right_layout->addWidget(apply_button, 0, 0);
   auto ok_button = new flat_button(tr("OK"));
   ok_button->setFont(generic_button_font);
   ok_button->set_style(generic_button_style);
   ok_button->set_hover_style(generic_button_hover_style);
   ok_button->set_focus_style(generic_button_focus_style);
   ok_button->connect_clicked_signal([=] { m_ok_signal(); });
   right_layout->addWidget(ok_button, 1, 0);
+  auto cancel_button = new flat_button(tr("Cancel"));
+  cancel_button->setFont(generic_button_font);
+  cancel_button->set_style(generic_button_style);
+  cancel_button->set_hover_style(generic_button_hover_style);
+  cancel_button->set_focus_style(generic_button_focus_style);
+  cancel_button->connect_clicked_signal([=] { m_cancel_signal(); });
```

Comment:
Shouldn't need to wrap the m_cancel_signal/m_ok_signal etc... in a lambda. Try just passing it directly like:

  button->connect_signal(m_cancel_signal);